src/main/java/com/pusher/client/connection/websocket/WebSocketConnection.java

```diff
@@ -42,6 +43,8 @@
     private volatile ConnectionState state = ConnectionState.DISCONNECTED;
     private WebSocketClientWrapper underlyingConnection;
     private String socketId;
+    private int reconnectAttempts = 0;
```

Comment:
This is never again set to `0`, which means that the client will have a connection, reconnect up to 6 times, get a new connection, and when it loses again, it won't retry.

Perhaps we should set it to `0` whenever we move to a `CONNECTED` state?